### DISCUSSION TITLE
App crash on startup at Entity Framework operations

### DISCUSSION THREAD

**Discussion by MAUIoxo** [dotnet/maui#24556] [Q&A]
I had a working App which I developed on Android/iOS Simulators and deployed successfully on my local iPhone. The App was working. Since May I really just did development (not much) with test on the Simulators.

Well, in July I wanted to deploy it to my iPhone again and couldn’t start it at all as it crashed instantly on startup.

I hat changes in NuGet versions and one of them in DevExpress which brought some problems with Trimming I had to figure out. Well, changed that, but the App still crashed. When I tried to run the Release App on Simulator I could get further as on my iPhone and figure out that it stopped on a simple Database access to SQLite via Entity Framework.

Anyway, I still needed to find out why it stopped working on my iPhone and as I tried different things with AOT compilation, Optimization, Trimming and nothing helped, I opened the following question on Stackoverflow: [Database Access with Entity Framework Core not Working anymore in Release](https://stackoverflow.com/questions/78904412/database-access-with-entity-framework-core-not-working-anymore-in-release-but-st)

Long story short: yesterday, I also wanted to prepare a small example project, made a copy of my existing App, renamed the App as well as changed the `<ApplicationTitle>`, `<ApplicationId>` and `<ApplicationIdGuid>`. I also renamed the the entire occurrence of the App name around the Solution, which basically were changes in the Namespace.

After doing that, I deployed to my iPhone and was really confused, as I could start the App this time and got over the Point where my Database access crashed the app previously and now it didn’t.

I thought, maybe my App is blocked by my iOS with the previous name or something like that. Could this be the case?

The App was released with the old Name before all the iOS Updates to 17.x which have had an impact? Also changes in Visual Studio 2022 Preview etc.

But, now changing the `<ApplicationTitle>`, `<ApplicationId>` and `<ApplicationIdGuid>` in the previous App to see if it was that didn’t bring success. So, compared to the copy of the App, renaming all namespaces and names inside the solution was missing. The original App still crashed with the changed parts above!

Now, my question is, how can this be explained and is there anything I have to change or do on Mac side which is used for compilation of the App or on iOS side where the App was deployed in developer mode on previous iOS versions?

@rolfbjarne - sorry for tagging you here explicitly, but I know that you are a real expert in such things and might have an answer?

My concern is that when I do not know what the reason was it stopped working, I might have problems later on and be stuck. Right now, it took me several weeks to even get to this point. Things were working before and from one to another it stopped and crashed right away on my iPhone, even without exceptions. Also no problems in Debug, but crashed without Exception on iPhone or even in Release on Simulator.

What I also find strange is, that when I deploy my original App on my iPhone (the one that crashes) my App size shows 225 MB in size where the working one shows 179 MB. I also deleted it several times and re-deployed with a fresh build. Still higher App size

**Visual Studio 2022**: Preview or release
**Maui**:  8.0.70, 8.0.80, 8.0.82
**iPhone**: 14 Pro Max
**iOS**:  17.6.1
**Xcode**:  15.4
**MacOS**: Sonoma 14.6.1 (latest Updates)

**Comment by gautambjain** [comment]
I had similar issue. I read at many places that this kind of thing can happen due to missing assemblies/plugins. And I could fix it from the steps [described here](https://github.com/AdamEssenmacher/GoogleApisForiOSComponents?tab=readme-ov-file#long-path-issue-workarounds). Basically we need to clean up rebuild again. But it is more than project cleanup.

How did I find those errors regarding missing assemblies/plugins? Go to Xcode / Devices and Simulators, select your connected iPhone and go to device log. When launching your app, it will show the errors in the device log.

I did the below in Visual Studio Code on Mac. You will need to do similar in Visual Studio 2022.

1. Close the project that's open in the IDE.
2. Run the command: **dotnet nuget locals all --clear**
3. Delete the bin and obj folders under your project folder.
4. Run the command **dotnet restore** (In my case just rebuilding all did the restore I believe)
5. Try running the app on the physical device. It should work.

If it doesn't work, make sure you check the device log for errors and get deeper into it.

Good luck!

**Reply by MAUIoxo** [reply]
First of all, thank you very much for your response, really appreciate any hint!

What I did before almost always, was deleting the NuGet Cache (C:\Users\<user>\.nuget\packages), bin/obj und also cleaning via CCleaner on my Mac. Then rebuilt and deployed.

What you described had to do with name lengths and also a good idea to check. I added the registry Key as described in this post.

Unfortunately, this did not help - same behavior.

I also checked a different approach I found somewhere and deleted my App on the iPhone and then restarted my iPhone which I did not do after deletion before. This also did not help. But, interestingly the App size now is different and it is just 209 MB

**Comment by rolfbjarne** [comment]
> @rolfbjarne - sorry for tagging you here explicitly, but I know that you are a real expert in such things and might have an answer?

You really need to figure out _why_ it's crashing, instead of applying solutions at random.

The first two things to look at are:

1. If there are any crash reports: https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#crash-reports
2. If anything is printed to the device log: https://support.apple.com/en-in/guide/console/cnsl1012/mac

That will hopefully give an idea about what the problem is.

**Reply by MAUIoxo** [reply]
Well, you are definitively right, the _why_ part is the interesting part also for me. But, I am really now at a point, I have absolutely no clue anymore.

**Small retrospection**:
Fact is that I had a working App and running both on Debug/Release on my iPhone. Then something changed, be it the iOS Update to 17.x, VS Update, .. that broke. I had a hard time to get to this point where I am right now. During that route there were even things like `Preferences.Get(Constants.ApplicationLanguage, "en");` that were not accepted anymore and just crashed the App (just closed it) until I found, that this should be changed to `Preferences.Default.Get(Constants.ApplicationLanguage, "en");` so that I can get over this crash. 

Also things like getting the `CultureInfo.InstalledUICulture` didn't want to get called anymore on iOS in Release and just crashed the App:
```cs
// Retrieve the installed UI Culture of the system
CultureInfo systemCulture = CultureInfo.InstalledUICulture;

// Extract the two-letter ISO language code from the UI Culture
string systemLanguageCode = systemCulture.TwoLetterISOLanguageName;
```

This I solved by creating a certain Platform Service which worked when called on iOS:

```cs
public string GetCurrentSystemLanguage()
{
    var systemLanguage = NSLocale.PreferredLanguages[0];
    var currentCulture = new CultureInfo(systemLanguage);

    return currentCulture.TwoLetterISOLanguageName;
}
```

These things were really strange to me as they worked before on my iPhone.

Last things that just crashed were related to a Database query with the Entity Framework working on a SQLite database file in `FileSystem.AppDataDirectory`. The Database uses a Migration on startup. When the app is started, I change to a certain `View` and inside the Constructor of the `ViewModel` I just have a simple call like `foodItems = _databaseService.DatabaseContext.Foods.OrderBy(foodItem => foodItem.Name).ToList();`.

This statement just crashes the App - the App is simply closed! In Debug it is working fine, in Release it closes the App.

Then came the "random" part you mentioned. When I tried to prepare a simple example project and renamed every .csproj, `<ApplicationTitle>`, `<ApplicationId>` and `<ApplicationIdGuid>`and Namespace of the copy of my original program, I did a Deployment on my iPhone and there it obviously it didn't crash. I can't say why, just that it didn't. Try-Catch around that statement did not help in the unrenamed project.

I also looked at the logs which I attached and looked at the outputs from the following method

```cs
private async void LoadAvailableFoodItemsAsync()
{
    Console.WriteLine("******* LoadAvailableFoodItemsAsync reached ***************");
    await App.Current.MainPage.DisplayAlert(AppResources.AppTitle, $"Davor: {call}", AppResources.OKButtonText);
    Console.WriteLine("******* After Display alert ***************");
    Console.WriteLine("******* before _databaseService.DatabaseContext.Foods.OrderBy(...) ***************");
    Console.WriteLine($"******* DatabaseService: {_databaseService} ***************");
    Console.WriteLine($"******* DatabaseService.Context: {_databaseService.DatabaseContext} ***************");
    Console.WriteLine($"******* DatabaseService.Context.Foods: {_databaseService.DatabaseContext.Foods} ***************");
    var foodItems = _databaseService.DatabaseContext.Foods.OrderBy(foodItem => foodItem.Name).ToList();
    Console.WriteLine("******* after _databaseService.DatabaseContext.Foods.OrderBy(...) ***************");
    
    await App.Current.MainPage.DisplayAlert(AppResources.AppTitle, $"after {call}", AppResources.OKButtonText);
    await App.Current.MainPage.DisplayAlert(AppResources.AppTitle, $"Element: {foodItems}", AppResources.OKButtonText);
    //AvailableFoodItems.ReplaceRange(foodItems);
    await App.Current.MainPage.DisplayAlert(AppResources.AppTitle, $"very end and after {call}", AppResources.OKButtonText);

    call++;
}
``` 

After `******* DatabaseService.Context.Foods: {_databaseService.DatabaseContext.Foods} ***************` I do not understand the messages and why it crashes.

[Logs_AppCrashing.txt](https://github.com/user-attachments/files/16846448/Logs_AppCrashing.txt)
[Logs_AppCrashing2.txt](https://github.com/user-attachments/files/16846454/Logs_AppCrashing2.txt)
[Logs_AppRenamed.txt](https://github.com/user-attachments/files/16846545/Logs_AppRenamed.txt)

**Reply by rolfbjarne** [reply]
> [Logs_AppRenamed.txt](https://github.com/user-attachments/files/16846545/Logs_AppRenamed.txt)

says "killed from app switcher", so that's not really a crash.

> Logs_AppCrashing.txt

This one is weird:

> Now flagged as pending exit for reason: workspace client connection invalidated

I don't think I've seen this before.

Are there any crash reports in Xcode? (https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#crash-reports)

> Logs_AppCrashing2.txt

This file is identical to the "Logs_AppCrashing.txt"

**Reply by MAUIoxo** [reply]
The resports were exported from the Console App and I filtered everything by the App Name. There, I have 2 tabs - one is for all messages and the other one just for errors and problems.

I also attached Logs_JustErrorsFilteredForApp.txt for another run and corresponding messages tab content. Exported just by selecting everything (Command + A) and then via Action > Copy and then I pasted it into the *.txt files. Don't know, if there is a better way. The problem only is there are a lot of messages coming in, especially when you additionally click on "Start". Therefore, just filtered by the App Name

[Logs_AppCrashing3.txt](https://github.com/user-attachments/files/16852985/Logs_AppCrashing3.txt)
[Logs_JustErrorsFilteredForApp.txt](https://github.com/user-attachments/files/16852952/Logs_JustErrorsFilteredForApp.txt)


For the reports in XCode 15.4 it looks just a little bit different and I hope what I did was correct. I actually opened XCode, had to create a dummy project to see the Window menu item (or press Command + Shift + 2). In that I was in directory `/Users/mauioxo/Library/Developer/Xcode/DeviceLogs/MauiOxo’s iPhone-00008120-000275A41A68201E`. From there, I exported the following directory in DeviceLogs.zip file

[Device Logs.zip](https://github.com/user-attachments/files/16852922/Device.Logs.zip)

If I can support with more info, just let me know. Strange for me still is that in Debug, I can step over the line that is the troublemaker right now. Do you think there might be a problem in the .csproj?

<br/>

**Hint**:  These .csproj also contain a lot of commented lines and show what flags I tried. But, I am still not sure which combination is the one that is recommended?

<br/> 

**.csproj of my Maui App**:
```xml
<Project Sdk="Microsoft.NET.Sdk">

	<PropertyGroup>
		<TargetFrameworks>net8.0-android;net8.0-ios</TargetFrameworks>
		<OutputType>Exe</OutputType>
        <!-- !!! Names are changed just for this discussion here !!! -->
		<RootNamespace>MyOptimizerApp</RootNamespace>
		<UseMaui>true</UseMaui>
		<SingleProject>true</SingleProject>
		<ImplicitUsings>enable</ImplicitUsings>
		<Nullable>enable</Nullable>
		<MauiVersion>8.0.82</MauiVersion>

		<!-- Display name -->
		<ApplicationTitle>My Optimizer App</ApplicationTitle>
		
		<!-- App Identifier -->
		<ApplicationId>com.yourcompany.MyOptimizerApp</ApplicationId>
		<ApplicationIdGuid>0f1568ac-b0d7-4202-838c-e20590586c24</ApplicationIdGuid>

		<!-- Versions -->
		<ApplicationDisplayVersion>1.0</ApplicationDisplayVersion>
		<ApplicationVersion>1</ApplicationVersion>

		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">14.2</SupportedOSPlatformVersion>
		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'android'">21.0</SupportedOSPlatformVersion>
	</PropertyGroup>

	<PropertyGroup Condition="'$(TargetFramework)'=='net8.0-ios'">
		<EnableAssemblyILStripping>false</EnableAssemblyILStripping>
		<CodesignKey>Apple Development: Created via API (8LTS3889TZ)</CodesignKey>
		<CodesignProvision>VS Codesign Provisioning Profile</CodesignProvision>
		<ProvisioningType>manual</ProvisioningType>
	</PropertyGroup>

	
	<!-- Disable Trimming for DevExpress on Android/iOS -->
	<PropertyGroup>
		<DXConfigureTrimming>false</DXConfigureTrimming>
	</PropertyGroup>

	<!-- Debug / Release configurations see -->
	<!-- https://docs.devexpress.com/MAUI/404979/common-concepts/build-configuration?utm_source=SupportCenter&utm_medium=website&utm_campaign=docs-feedback&utm_content=T1245248#devexpress-build-settings -->
	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|AnyCPU'">
		<UseInterpreter>true</UseInterpreter>
		<!--<MtouchInterpreter>all</MtouchInterpreter>-->
		<MtouchLink>SdkOnly</MtouchLink>
		<MtouchUseLlvm>false</MtouchUseLlvm>
	</PropertyGroup>
	
	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|AnyCPU'">
		<UseInterpreter>true</UseInterpreter>
		<!--<UseInterpreter>false</UseInterpreter>-->
		<Optimize>false</Optimize>
		<!--<RunAOTCompilation>true</RunAOTCompilation>-->
		<RunAOTCompilation>false</RunAOTCompilation>
		<MtouchLink>SdkOnly</MtouchLink>
		<MtouchUseLlvm>false</MtouchUseLlvm>


		<MtouchInterpreter>all</MtouchInterpreter>
		<!--<DebugSymbols>true</DebugSymbols>-->
		<!--<DebugType>portable</DebugType>-->
		<!--<Optimize>true</Optimize>-->
		<!--<IncludeSource>true</IncludeSource>-->
		<!--<RuntimeIdentifiers>ios-arm64;iossimulator-arm64</RuntimeIdentifiers>-->
	</PropertyGroup>
		
	<!-- Release Build Debugging Android -->
	<!--<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net8.0-android|AnyCPU'">
		<TargetFramework>net8.0-android34.0</TargetFramework>
		<SupportedOSPlatformVersion>21</SupportedOSPlatformVersion>
		<Optimize>false</Optimize>
		<DebugSymbols>true</DebugSymbols>
		<DebugType>portable</DebugType>
		<EmbedAssembliesIntoApk>false</EmbedAssembliesIntoApk>
		<AndroidPackageFormat>apk</AndroidPackageFormat>
	</PropertyGroup>-->

	<ItemGroup>
		<!-- App Icon -->
		<MauiIcon Include="Resources\AppIcon\appicon.svg" Color="#512BD4" />

		<!-- Images -->
		<MauiImage Include="Resources\Images\*" />
		<MauiImage Update="Resources\Images\dotnet_bot.png" Resize="True" BaseSize="300,185" />

		<!-- Custom Fonts -->
		<MauiFont Include="Resources\Fonts\*" />

		<!-- Raw Assets (also remove the "Resources\Raw" prefix) -->
		<MauiAsset Include="Resources\Raw\**" LogicalName="%(RecursiveDir)%(Filename)%(Extension)" />
	</ItemGroup>

	<ItemGroup>
		<Folder Include="Pages\HelpPages\" />
	</ItemGroup>

	<!-- Splashscreen for Android -->
	<ItemGroup>
		<None Update="Pages\Views\BottomSheetContentView.xaml">
		  <Generator>MSBuild:Compile</Generator>
		</None>
		<None Update="Resources\Splash\mo_splash.png" Color="#000000" BaseSize="1024,1024" Resize="false" />
		<None Update="Resources\Splash\splash.svg" Color="#512BD4" BaseSize="128,128" />
	</ItemGroup>
	
	<!-- Splashscreen for iOS -->
	<ItemGroup Condition="'$(TargetFramework)'=='net8.0-ios'">
		<InterfaceDefinition Include="LaunchScreen.storyboard" />
	</ItemGroup>

	<ItemGroup>
		<PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="8.0.8" />
		<PackageReference Include="Microsoft.Maui.Controls" Version="$(MauiVersion)" />
		<PackageReference Include="Microsoft.Maui.Controls.Compatibility" Version="$(MauiVersion)" />
		<PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="8.0.0" />
		<PackageReference Include="CommunityToolkit.Diagnostics" Version="8.3.0" />
		<PackageReference Include="CommunityToolkit.Maui" Version="9.0.3" />
		<PackageReference Include="CommunityToolkit.Mvvm" Version="8.3.0" />
		<PackageReference Include="Refractored.MvvmHelpers" Version="1.6.2" />
		<PackageReference Include="DevExpress.Maui.Core" Version="24.1.5" />
		<PackageReference Include="DevExpress.Maui.CollectionView" Version="24.1.5" />
		<PackageReference Include="LocalizationResourceManager.Maui" Version="1.2.1" />
		<PackageReference Include="Microsoft.EntityFrameworkCore" Version="8.0.8" />
		<PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="8.0.8">
			<PrivateAssets>all</PrivateAssets>
			<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
		</PackageReference>
		<PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="8.0.8" />
		<!--<PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite.Core" Version="8.0.8" />-->
		<PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="8.0.8">
			<PrivateAssets>all</PrivateAssets>
			<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
		</PackageReference>
		<PackageReference Include="Sharpnado.Tabs.Maui" Version="3.0.1" />
		<PackageReference Include="Sentry" Version="4.10.2" />
		<PackageReference Include="Sentry.Maui" Version="4.10.2" />
	</ItemGroup>

	<ItemGroup>
	  <ProjectReference Include="..\MyOptimizerDatabase\MyOptimizerDatabase.csproj" />
	  <ProjectReference Include="..\Maui.Controls.BottomSheet-main\src\Maui.Controls.BottomSheet\Maui.Controls.BottomSheet.csproj" />
	</ItemGroup>

	<ItemGroup>
	  <Compile Update="Platforms\Android\LocalizationService_Android.cs">
	    <ExcludeFromCurrentConfiguration>false</ExcludeFromCurrentConfiguration>
	  </Compile>
	  <Compile Update="Platforms\iOS\MauiProgram.iOS.cs">
	    <ExcludeFromCurrentConfiguration>false</ExcludeFromCurrentConfiguration>
	  </Compile>
	  <Compile Update="Resources\Localization\AppResources.Designer.cs">
	    <DesignTime>True</DesignTime>
	    <AutoGen>True</AutoGen>
	    <DependentUpon>AppResources.resx</DependentUpon>
	  </Compile>
	</ItemGroup>

	<ItemGroup>
	  <EmbeddedResource Update="Resources\Localization\AppResources.resx">
	    <Generator>ResXFileCodeGenerator</Generator>
	    <LastGenOutput>AppResources.Designer.cs</LastGenOutput>
	  </EmbeddedResource>
	</ItemGroup>

	<ItemGroup>
	  <MauiXaml Update="Pages\Help.xaml">
	    <Generator>MSBuild:Compile</Generator>
	  </MauiXaml>
	  <MauiXaml Update="Pages\Settings.xaml">
	    <Generator>MSBuild:Compile</Generator>
	  </MauiXaml>
	  <MauiXaml Update="Pages\Views\BottomSheetContentView.xaml">
	    <Generator>MSBuild:Compile</Generator>
	  </MauiXaml>
	  <MauiXaml Update="Pages\Views\Controls\CustomActivityIndicator\CustomCalculationActivityIndicator.xaml">
	    <Generator>MSBuild:Compile</Generator>
	  </MauiXaml>
	</ItemGroup>	

</Project>
```

**.csproj of my .NET Core Database DLL used to provide the Entity Framework Core DbContext and Model Classes**:
This .NET Core Project is linked in the .NET MAUI Project
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <!--<SignAssembly>False</SignAssembly>-->
	<SignAssembly>true</SignAssembly>
  </PropertyGroup>

  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
    <IsAotCompatible>True</IsAotCompatible>
    <IsTrimmable>False</IsTrimmable>
  </PropertyGroup>

  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
	<!--<UseInterpreter>false</UseInterpreter>-->
    <IsAotCompatible>true</IsAotCompatible>
	<!--<Optimize>true</Optimize>-->
    <IsTrimmable>false</IsTrimmable>
	<LinkerPleaseIncludeAssemblies>MyOptimizerDatabase</LinkerPleaseIncludeAssemblies>
	<MtouchLinkSkip>Microsoft.EntityFrameworkCore;Microsoft.EntityFrameworkCore.Sqlite;MyOptimizerDatabase;MyOptimizerApp.Database;MyOptimizerApp.DatabaseModels</MtouchLinkSkip>

	<UseInterpreter>true</UseInterpreter>
	<MtouchInterpreter>all</MtouchInterpreter>
	<MtouchUseLlvm>false</MtouchUseLlvm>
	  <!--<DebugSymbols>true</DebugSymbols>-->
	  <!--<DebugType>portable</DebugType>-->
	  <!--<Optimize>true</Optimize>-->
	  <!--<IncludeSource>true</IncludeSource>-->
  </PropertyGroup>

  <!-- Release Debugging Configuration -->
  <!--<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
    <IsAotCompatible>true</IsAotCompatible>
	<IsTrimmable>false</IsTrimmable>
	<Optimize>false</Optimize>
	<DebugSymbols>true</DebugSymbols>
	<DebugType>full</DebugType>
	<DebugType>portable</DebugType>
	<RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
  </PropertyGroup>-->

  <ItemGroup>
	  <RuntimeHostConfiguration Include="System.Globalization.Invariant" Value="true" />
  </ItemGroup>

  <ItemGroup>
	<PackageReference Include="CommunityToolkit.Mvvm" Version="8.3.0" />
	<PackageReference Include="Microsoft.EntityFrameworkCore" Version="8.0.8" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="8.0.8">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="8.0.8" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="8.0.8" />
    <!--<PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite.Core" Version="8.0.8" />-->
	<PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="8.0.8">
		<PrivateAssets>all</PrivateAssets>
		<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
	</PackageReference>
  </ItemGroup>

</Project>
```

**Reply by kunj-nilkanth** [reply]
I am facing a simmilar kind of issue, where ony of my app which is already live on appstore that works on other ios versions but when in open it on iOS 17.6.1 it doesn;t even open the splash and it crashes, but when i run the same code from XCode with release mode it works. i can not understand what the issue is.

**Reply by rolfbjarne** [reply]
> From there, I exported the following directory in DeviceLogs.zip file

That looks right, but there are no relevant crash reports, so looks like that was a dead end.

> Do you think there might be a problem in the .csproj?

Nothing stands out from the csprojs.

> The resports were exported from the Console App and I filtered everything by the App Name.

Can you try not filtering? Unfortunately I can't find anything in the filtered logs telling me what's going on :/

**Reply by MAUIoxo** [reply]
Exactly, it was the same for me. It basically crashed right after the splash screen. Since I had customized it, I suspected that might have been the cause.

At some point, I started to rebuild my app step by step in a new project.

That’s when I realized that iOS (if that's the reason) is quite unforgiving when it comes to even the smallest 'mistakes' and just crashes.

Then I tried to debug a release build on the iOS simulator with the part from my posted .csproj, but I couldn’t proceed with the database-related parts because the app simply closed on the simulator without throwing an exception or anything.

However, on the simulator, it failed to proceed beyond certain newly added statements related to the database (in my case), but on the iPhone, they worked fine.

At this point, I can't even debug the release build in the simulator anymore, as the app closes before that. On the iPhone, at least I can reach the mentioned problematic part.

Unfortunately, I’m stuck and can't fix it without help. I’ve swapped libraries, even gone back quite a bit, but the crashes kept happening. As I mentioned, it’s quite funny that after renaming something, it suddenly passed that problematic code section again.

I also wanted to investigate further, if possible, whether it could be related to my Entity Framework Core data model or DbContext, because I noticed something strange. I registered my DatabaseContext in the MauiProgram.cs file and tried to access it in the ViewModel through constructor dependency injection to call Migrate(), which caused a crash as well.

But through my own DatabaseService, which holds the DbContext, it worked.

Everything is a bit confusing, and @rolfbjarne, that’s why this random trial-and-error approach continues."

Let me know if you need any further changes!

**Reply by rolfbjarne** [reply]
If you can create a minimal test project, you can file an issue at https://github.com/xamarin/xamarin-macios/issues/new/choose (attaching/linking to the project) and I'll have a look.

**Reply by MAUIoxo** [reply]
I can try to do that - is the „Running an Application“ the right section or „Maui“? I would try to create such a project and protect some parts if openly visible

**Reply by rolfbjarne** [reply]
Choose "Running an Application"

**Reply by MAUIoxo** [reply]
Here is the logs from today which should provide more information. The second file is the export just from the errors and problems tab within the Console App:

[Logs_AppCrashing4.zip](https://github.com/user-attachments/files/16884188/Logs_AppCrashing4.zip)
[Logs_JustErrorsCrashing4.txt](https://github.com/user-attachments/files/16884189/Logs_JustErrorsCrashing4.txt)

Providing a sample program will definitively take some time... Hope, the files above can also provide some more information

**Reply by MAUIoxo** [reply]
It behaves as if it doesn't like the name I'm using. As soon as I start with a copy of my project (to prepare a sample project) and start renaming the .csproj of my database assembly in my Solution, along with the corresponding namespaces (e.g. changed from DataOptimizerApp to DataOptimizer), I can deploy and run it without issues. But when I rename everything back, it stops working again.

Is there something like a blacklist in iOS that locks out something, or could the renaming somehow affect how my SQLite database works (or doesn't work)? And the other observation is that everytime it works with a copy, the size for the App on my iPhone is way smaller. As if it uses an old copy of the database or the App folder additionally copies also older DLLs or so if not renamed - this obersavtion is also very strange and I do not have a proper explanation for that => sorry for guessing all that

**Reply by rolfbjarne** [reply]
Does it start working again if you manually delete the app from your device first?

**Reply by rolfbjarne** [reply]
> Logs_AppCrashing4.zip

Unfortunately there's nothing interesting in this log either :/

I'm afraid that means I'll need a test project to reproduce, I'm out of ideas.

**Reply by MAUIoxo** [reply]
It does not start working again when I manually delete the App first. I also did it like: Delete the App, Shut Down iOS and Re-Start the iPhone again, so that the Caches should be cleared. Then I deployed again... And there, deploying the App with that naming is higher in size...

**Reply by MAUIoxo** [reply]
> I am facing a simmilar kind of issue, where ony of my app which is already live on appstore that works on other ios versions but when in open it on iOS 17.6.1 it doesn;t even open the splash and it crashes, but when i run the same code from XCode with release mode it works. i can not understand what the issue is.

What is the plan with your app? That would be the worst case, having people downloading an App and after a certain Update nothing works anymore and just crashes. Any idea how you will proceed facing this issue?

**Reply by MAUIoxo** [reply]
> > Logs_AppCrashing4.zip
> 
> Unfortunately there's nothing interesting in this log either :/
> 
> I'm afraid that means I'll need a test project to reproduce, I'm out of ideas.

If you give me some time, I‘ll try to prepare and reduce it as far as I can

**Reply by MAUIoxo** [reply]
@rolfbjarne - Error report is done. I hope you can find all the information you need. If not, just let me know, what I should add

**Reply by rolfbjarne** [reply]
@MAUIoxo I'm not sure I understand; I need a test project to diagnose this particular problem, not an error report.

**Reply by MAUIoxo** [reply]
Ah, it was misleading 😅 What I actually wanted to say was that I wrote an error report including the minimal reproducible example in your Xamarin project platform (https://github.com/xamarin/xamarin-macios/issues/new/choose).

I placed it under the category “Running an Application” as you proposed.

You can have a look at the (obfuscated) example 😊 (sorry for the confusion):

[Xamarin issue #21228](https://github.com/xamarin/xamarin-macios/issues/21228)

**Comment by rolfbjarne** [comment]
Closing as per https://github.com/xamarin/xamarin-macios/issues/21228.
